Emoji font

999eagle opened the pull request:
This PR adds a font for emoji and changes some text styles to fall back to this font. That way, missing glyphs in the main font (for example emoji) will be rendered using the emoji font.
The font used is EmojiOne from https://github.com/maximbaz/emojione-fonts, with that it's under a CC-BY-4.0 license so it needs attribution to use.

lifenautjoe commented:
Hmmm how does this behave on mobile phones 🤔 . I'll do some testing. 

Also on emojione, it is free license but with some very harsh limitations https://github.com/joypixels/emojione-assets/blob/master/LICENSE.md , so we've gotta take a look at those too before merging.

999eagle commented:
Yeah license does require attribution to use. It's also quite large (about 18 MiB - even after zip compression as used in APKs) so this might not be the best option.

999eagle commented:
Another option might be to use Google's NotoColorEmoji font, licensed under the SIL Open Font License, available for download here: https://www.google.com/get/noto/#emoji-zsye-color

uiboy commented:
So maybe we could use this https://github.com/eosrei/twemoji-color-font

999eagle commented:
I've tried the twemoji font now, but sadly the color format used there (SVG in OT) isn't supported on Android and thus emoji are displayed as outlines only instead of full color glyphs.
See the attached image for an example of how this looks in text.
![image](https://user-images.githubusercontent.com/1221984/56805209-d050bf00-6817-11e9-850b-24b8a95dc17d.png)

999eagle commented:
I've rebased this onto the current `develop` branch and replaced the font with Twemoji. This font was built in Google's CBDT/CBLC format and displays colored emojis on Android, but I couldn't test this on iOS yet.

Mennaruuk commented:
Twemoji is also CC-BY 4.0, so an attribution is required. They, however, mention that they "will accept a mention in a project README or an 'About' section or footer on a website. In mobile applications, a common place would be in the Settings/About section (for example, see the mobile Twitter application Settings->About->Legal section). We would consider a mention in the HTML/JS source sufficient also." @999eagle 

[Source](https://twemoji.twitter.com/)
